packages/flame/lib/src/camera/viewport.dart

```diff
@@ -54,15 +55,24 @@ abstract class Viewport extends Component
   /// Changing the size at runtime triggers the [onViewportResize] event. The
   /// size cannot be negative.
   @override
-  Vector2 get size => _size;
+  Vector2 get size {
+    if (!_isInitialized && camera.parent is FlameGame) {
+      // This is so that the size can be accessed before the viewport is fully
+      // mounted.
+      onGameResize((camera.parent! as FlameGame).canvasSize);
```

Comment:
It would get an extra `onGameResize`, but only the viewport will get this, if it has any children they haven't been mounted yet. It's a little ugly, but I don't know of a better way to solve this.